app/models/ranking/api_client.rb

```diff
@@ -0,0 +1,66 @@
+require 'googleauth/stores/redis_token_store'
+require 'google/apis/analyticsreporting_v4'
+
+class Ranking::ApiClient
+  Analytics = Google::Apis::AnalyticsreportingV4
+
+  def initialize()
+    if use_api?
+      @client = Analytics::AnalyticsReportingService.new
+
+      scope = ['https://www.googleapis.com/auth/analytics.readonly']
+      @client.authorization = Google::Auth::ServiceAccountCredentials.make_creds(
+        json_key_io: json_key_io,
+        scope: scope
+      )
+    end
+  end
+
+  def popular_posts(options)
+    if use_api?
+      popular_posts_data_from_google_api(options)
+    else
+      popular_posts_data_from_local_file
+    end
+  end
+
+  private
+
+  def use_api?
+    Rails.application.config.x.google_api_client_email && Rails.application.config.x.google_api_private_key
+  end
+
+  def popular_posts_data_from_google_api(options)
```

Comment:
Please use [keyword args](https://robots.thoughtbot.com/ruby-2-keyword-arguments) instead of `options`.